### PR TITLE
Add LOG_RECEIVED to customize logging

### LINKED ISSUE #6756: Ability to customize `Received task` message
- [X] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Enhancement%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical enhancement to an existing feature.
- [X] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22Issue+Type%3A+Enhancement%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed enhancements.
- [X] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the if the same enhancement was already implemented in the
  master branch.
- [X] I have included all related issues and possible duplicate issues in this issue
      (If there are none, check this box anyway).

#### Related Issues

- None

#### Possible Duplicates

- None

# Brief Summary
Currently `LOG_SUCCESS` and friends allows customizing the messages, but `Received task` is fixed. It would be good for this to be customizable and include the same values in `extra` (i.e. use `celery.app.trace.info` rather than `logging.info`). I was thinking something like this:
```
info('Received task: %s', {
    'name': req.name,
    'id': req.id,
})
```

# Design

## Architectural Considerations
Should be a non-breaking change that only adds a customization point

## Proposed Behavior
Add `LOG_RECEIVED` so it can be customized as well

## Proposed UI/UX
N/A

## Diagrams
N/A

## Alternatives
None that I can think of

### PR DESCRIPTION
**Pull request by daveisfera**
## Description

Add `LOG_RECEIVED` so `Received task` logging can be customized like `LOG_SUCCESS` and friends.

Fixes #6756

### CODE REVIEW

**Review by auvipy — CHANGES_REQUESTED** [review]
can you add some tests for the change?

**Comment by daveisfera** [comment]
Can you point me to where/how I'd do that. I don't see any tests for `LOG_SUCCESS` and friends, so I'm not sure where I'd even start.

**Review by thedrow — APPROVED** [review]
This is not tested and is probably minor enough not to warrant a test.
LGTM.

**Review by maybe-sybr — CHANGES_REQUESTED** [review]
LGTM with only a style nit to pick. We could land this in 5.1 if @auvipy and/or @thedrow are comfortable doing so before RC1 IMO :)

**Comment by maybe-sybr** [comment]
> Can you point me to where/how I'd do that. I don't see any tests for `LOG_SUCCESS` and friends, so I'm not sure where I'd even start.

AFAICT there are no current tests and this is in a kind of annoying spot to test easily. I'd imagine a test a bit like:
https://github.com/celery/celery/blob/6dd385258297c89843bfe73299e5f7eebf0e98e2/t/unit/worker/test_strategy.py#L168-L175
would be necessary but I think without an existing example it would be difficult for @daveisfera to work out exactly what to do.

**Comment by auvipy** [comment]
In general I am not against small enhancements O:)

File: celery/app/trace.py [hunk 1]
```diff
@@ -52,6 +52,11 @@
 
 logger = get_logger(__name__)
 
+#: Format string used to log task receipt.
+LOG_RECEIVED = """\
+Received task %(name)s[%(id)s]\
```
[the remark this answers is not in the document]
Comment (maybe-sybr) [reply celery/app/trace.py]: I've just done this in a squashable commit

**Comment by maybe-sybr** [comment]
I just pushed a squashable commit which does a bit of cleanup as well as a couple of changes which should make this more usable IMO:
 1. Because `LOG_RECEIVED` was being from imported, overriding `celery.app.trace.LOG_RECEIVED` wouldn't actually do anything AFAICT
   * the `app.trace` module is now imported instead
 2. `app.trace` defined a helper for doing `info` logging which was being imported and used but that would mean the logger to which we emit received messages too would change
   * I've changed that back so it uses the `logger` defined in `worker.strategy` again

I'm just about to run up some tests for my other work which should let me smoke test this in a dev testing environment. And here's some output from an old test case I had lying around:

* Without an override:
```
[2021-05-11 10:51:05,971: INFO/MainProcess] Task __main__.prep_work_before_retry[54c683cb-dd54-4858-a929-df6ba9b22117] received
[2021-05-11 10:51:05,971: INFO/ForkPoolWorker-6] Task __main__.retry_once[805916ec-c024-435d-a3dc-42b2ab9119eb] ignored
[2021-05-11 10:51:05,972: INFO/ForkPoolWorker-7] Task __main__.retry_once[7919126e-20e4-4d98-b904-450582f32415] ignored
[2021-05-11 10:51:05,972: INFO/MainProcess] Task __main__.prep_work_before_retry[11cdce0c-d1ba-4aff-b0d6-f1222f206875] received
[2021-05-11 10:51:05,992: INFO/MainProcess] Task __main__.retry_once[7919126e-20e4-4d98-b904-450582f32415] received
[2021-05-11 10:51:05,993: INFO/ForkPoolWorker-1] Task __main__.prep_work_before_retry[11cdce0c-d1ba-4aff-b0d6-f1222f206875] succeeded in 0.020189827999274712s: 1337
[2021-05-11 10:51:05,993: INFO/ForkPoolWorker-8] Task __main__.prep_work_before_retry[54c683cb-dd54-4858-a929-df6ba9b22117] succeeded in 0.020225143001880497s: 1337
[2021-05-11 10:51:05,993: INFO/MainProcess] Task __main__.retry_once[805916ec-c024-435d-a3dc-42b2ab9119eb] received
[2021-05-11 10:51:05,995: INFO/MainProcess] Task __main__.do_something[d0d8eb77-980c-484a-9998-a8dbb41145b4] received
[2021-05-11 10:51:05,995: INFO/ForkPoolWorker-6] Task __main__.retry_once[7919126e-20e4-4d98-b904-450582f32415] succeeded in 0.0018187710011261515s: 31337
[2021-05-11 10:51:05,995: INFO/ForkPoolWorker-7] Task __main__.retry_once[805916ec-c024-435d-a3dc-42b2ab9119eb] succeeded in 0.0019330780050950125s: 31337
[2021-05-11 10:51:05,996: INFO/MainProcess] Task __main__.do_something[b5e6a64f-98c1-4706-8795-8b3a66ea6686] received
[2021-05-11 10:51:05,996: INFO/ForkPoolWorker-8] Task __main__.do_something[d0d8eb77-980c-484a-9998-a8dbb41145b4] succeeded in 0.00042758700146805495s: 42
[2021-05-11 10:51:05,997: INFO/ForkPoolWorker-6] Task __main__.do_something[b5e6a64f-98c1-4706-8795-8b3a66ea6686] succeeded in 0.0004646769957616925s: 42
```

* With the following line: `celery.app.trace.LOG_RECEIVED = "CUSTOM CUSTOM: Task %(name)s (%(id)s)"`
```
[2021-05-11 10:52:18,023: INFO/MainProcess] CUSTOM CUSTOM: Task __main__.prep_work_before_retry (9f28ef04-0025-4118-9e34-26a7bb180e73)
[2021-05-11 10:52:18,023: INFO/ForkPoolWorker-6] Task __main__.retry_once[824bb8ad-cfb4-4951-a7d4-68795d15dab2] ignored
[2021-05-11 10:52:18,034: INFO/MainProcess] CUSTOM CUSTOM: Task __main__.retry_once (824bb8ad-cfb4-4951-a7d4-68795d15dab2)
[2021-05-11 10:52:18,034: INFO/ForkPoolWorker-7] Task __main__.prep_work_before_retry[9f28ef04-0025-4118-9e34-26a7bb180e73] succeeded in 0.010441186001116876s: 1337
[2021-05-11 10:52:18,036: INFO/MainProcess] CUSTOM CUSTOM: Task __main__.do_something (8f7d5df6-2ab7-4141-b828-8a774f7dc2f0)
[2021-05-11 10:52:18,036: INFO/ForkPoolWorker-6] Task __main__.retry_once[824bb8ad-cfb4-4951-a7d4-68795d15dab2] succeeded in 0.001818830001866445s: 31337
[2021-05-11 10:52:18,036: INFO/ForkPoolWorker-7] Task __main__.do_something[8f7d5df6-2ab7-4141-b828-8a774f7dc2f0] succeeded in 0.0003603869990911335s: 42
```

lgtm

**Comment by daveisfera** [comment]
> 1. Because `LOG_RECEIVED` was being from imported, overriding `celery.app.trace.LOG_RECEIVED` wouldn't actually do anything AFAICT
> 
> * the `app.trace` module is now imported instead

It worked in my testing, but I'm totally fine with that change.

> 2. `app.trace` defined a helper for doing `info` logging which was being imported and used but that would mean the logger to which we emit received messages too would change
> 
> * I've changed that back so it uses the `logger` defined in `worker.strategy` again

Oh ya, definitely didn't mean to change that, so good catch there, but I was wasn't do get the `extra={'data': context}`, so could that be added in to the actual call?

**Review by auvipy — COMMENTED** [review]
the changes need some doc updates IMHO

**Comment by maybe-sybr** [comment]
> > 1. Because `LOG_RECEIVED` was being from imported, overriding `celery.app.trace.LOG_RECEIVED` wouldn't actually do anything AFAICT
> > 
> > 
> > * the `app.trace` module is now imported instead
> 
> It worked in my testing, but I'm totally fine with that change.

huh, weird. Oh well as long at it's working now I suppose :)

> 
> > 1. `app.trace` defined a helper for doing `info` logging which was being imported and used but that would mean the logger to which we emit received messages too would change
> > 
> > 
> > * I've changed that back so it uses the `logger` defined in `worker.strategy` again
> 
> Oh ya, definitely didn't mean to change that, so good catch there, but I did want to [get the `extra={'data': context}`](https://github.com/celery/celery/blob/5d6000987423b1259d0e69b3805bffc8a6cda353/celery/app/trace.py#L131), so could that be added in to the actual call?

I see no problem with that, it'll make all of the log lines at info level using these format strings from `app.trace` look the same to a handler capturing both loggers. I'll push another change for that now.

**Comment by maybe-sybr** [comment]
> the changes need some doc updates IMHO

Yeah, I can't spot any existing documentation about these format strings unfortunately. They got added in a pretty broad refactor commit (1e9dd26592). I think we could potentially add a logging customisation section to `docs/userguide/extending.rst`, or make a new, more complete `docs/userguide/logging.rst` document.

If we go for a little section in `extending.rst`, it could be done in this PR but it'd probably be worth pushing it back to 5.1.x or 5.2 to avoid blocking the RC/release. If we want something more complete about the logging subsystem, I think we should take that on notice and make a new issue to track the effort.

**Comment by auvipy** [comment]
> If we go for a little section in `extending.rst`, it could be done in this PR

this can be a good start

**Review by auvipy — APPROVED** [review]

**Comment by maybe-sybr** [comment]
> > If we go for a little section in `extending.rst`, it could be done in this PR
> 
> this can be a good start

Just added a sub-section under the worker section of `extending.rst` for customising task logging

**Comment by maybe-sybr** [comment]
And added a couple of tests for this. I think this will fix up coverage and give us an example of how to test the rest of the log overrides in a potential future change.

**Review by auvipy — APPROVED** [review]

**Pull request merged**